### PR TITLE
Push KrakenD User-Agent only if not overriding forwarded Header

### PR DESCRIPTION
**Pull request by PuKoren**
This PR prevent KrakenD from overriding the User-Agent, if the user wants to forward that header (if he specifies `User-Agent` or `*` and the header is present)

### CODE REVIEW

File: router/gin/endpoint.go [hunk 1]
```diff
@@ -109,8 +109,13 @@ func NewRequest(headersToSend []string) func(*gin.Context, []string) *proxy.Requ
 				headers[k] = h
 			}
 		}
+
 		headers["X-Forwarded-For"] = []string{c.ClientIP()}
-		headers["User-Agent"] = router.UserAgentHeaderValue
+		// if User-Agent is not forwarded using headersToSend, we set
+		// the KrakenD router User Agent value
+		if headers["User-Agent"] == nil {
```
Comment (kpacha) [inline router/gin/endpoint.go]: I'd rather check just if the value exists or not
```
if _, ok := headers["User-Agent"]; !ok {
```

File: router/gin/endpoint.go [hunk 2]
```diff
@@ -109,8 +109,13 @@ func NewRequest(headersToSend []string) func(*gin.Context, []string) *proxy.Requ
...
+			headers["User-Agent"] = router.UserAgentHeaderValue
+		}
```
Comment (kpacha) [inline router/gin/endpoint.go]: also, I think we should let the backends know the request has been forwarded by the krakend, so I'd add an else statement
```
} else {
	headers["X-Forwarded-Via"] = router.UserAgentHeaderValue
}
```

Comment (PuKoren) [reply router/gin/endpoint.go]: Would it makes sense to always add it (not only in the `else`)?

Comment (kpacha) [reply router/gin/endpoint.go]: If the `User-Agent` already contains the `UserAgentHeaderValue` I think we can't avoid the duplication

**Comment by PuKoren** [comment]
@kpacha I made the requested changes, thanks for your review!

While doing it however I felt a little bit weird that sometimes info is found in `User-Agent`, sometimes in `X-Forwarded-Via`. Maybe it would be simpler to always fill the `X-Forwarded-Via` and not use `User-Agent`, but it may be a breaking change.

**Comment by kpacha** [comment]
@PuKoren I understand your concern but I think we should assume that having the `User-Agent` header in the list of headers to pass is something the backend requires and, therefore, its metrics and log collectors and analyzers will be aware of that point

**Pull request merged**